scripts/sync-kernel.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+usage () {
+    echo "USAGE: ./sync-kernel.sh <kernel-repo> <libbpf-repo> [<baseline-commit>]"
+    echo ""
+    echo "If <baseline-commit> is not specified, it's read from <libbpf-repo>/CHECKPOINT-COMMIT"
+    exit 1
+}
+
+LINUX_REPO=${1-""}
+LIBBPF_REPO=${2-""}
+
+if [ -z "${LINUX_REPO}" ]; then
+    usage
+fi
+if [ -z "${LIBBPF_REPO}" ]; then
+    usage
+fi
+
+set -eu
+
+WORKDIR=$(pwd)
+trap "cd ${WORKDIR}; exit" INT TERM EXIT
+
+echo "WORKDIR:     ${WORKDIR}"
+echo "LINUX REPO:  ${LINUX_REPO}"
+echo "LIBBPF REPO: ${LIBBPF_REPO}"
+
+SUFFIX=$(date --utc +%Y-%m-%dT%H-%M-%S.%3NZ)
+BASELINE_COMMIT=${3-$(cat ${LIBBPF_REPO}/CHECKPOINT-COMMIT)}
+
+# Use current kernel repo HEAD as a source of patches
+cd ${LINUX_REPO}
+CHECKPOINT_COMMIT=$(git rev-parse HEAD)
+git branch libbpf-${SUFFIX} HEAD
+
+echo "SUFFIX: ${SUFFIX}"
+echo "BASELINE COMMIT: ${BASELINE_COMMIT}"
+echo "CHECKPOINT COMMIT: ${CHECKPOINT_COMMIT}"
+
+# Squash state of kernel repo at baseline into single commit
+SQUASHED_BASELINE_COMMIT=$(git commit-tree ${BASELINE_COMMIT}^{tree} -m "BASELINE ${BASELINE_COMMIT}")
+echo "SQUASHED BASELINE COMMIT: ${SQUASHED_BASELINE_COMMIT}"
+
+# Re-apply the rest of commits
+git rebase --onto ${SQUASHED_BASELINE_COMMIT} ${BASELINE_COMMIT} libbpf-${SUFFIX}
+
+# Move all libbpf files into libbpf-projection directory
+git filter-branch --prune-empty -f --tree-filter '                                                                             \
```

Comment:
didn't know such git command exist :)

Reply:
it's a "with great power comes great responsibility" type of command :)